Compression/Compression.cs

```diff
@@ -874,5 +875,44 @@ public static IEnumerable<string> GetZipEntryFileNames(string zipFileName)
                 }
             }
         }
+
+        /// <summary>
+        /// Extracts a 7-zip archive to disk, using the 7-zip CLI utility
+        /// </summary>
+        /// <param name="inputFile">Path to the 7z file</param>
+        /// <param name="outputDirectory">Directory to output contents of 7z</param>
+        /// <param name="execTimeout">Timeout in seconds for how long we should wait for the extraction to complete</param>
+        /// <exception cref="Exception">The extraction failed because of a timeout or the exit code was not 0</exception>
+        public static void Extract7ZipArchive(string inputFile, string outputDirectory = null, int execTimeout = 60)
+        {
+            outputDirectory = outputDirectory ?? $"{Guid.NewGuid()}";
+
+            var zipper = OS.IsWindows ? "C:/Program Files/7-Zip/7z.exe" : "7z";
+            var psi = new ProcessStartInfo(zipper, " e " + inputFile + " -o" + outputDirectory)
+            {
+                CreateNoWindow = true,
+                WindowStyle = ProcessWindowStyle.Hidden,
+                UseShellExecute = false,
+                RedirectStandardOutput = true
+            };
+
+            var process = new Process();
+            process.StartInfo = psi;
+            process.Start();
+
+            while (!process.StandardOutput.EndOfStream)
+            {
+                process.StandardOutput.ReadLine();
```

Comment:
Since this isn't being used suggest we just remove this and set `RedirectStandardOutput` to false?